plugins/inputs/snmp/snmp_test.go

```diff
@@ -287,44 +287,30 @@ func TestGetSNMPConnection_v2(t *testing.T) {
 }
 
 func TestGetSNMPConnectionTCP(t *testing.T) {
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go stubTCPServer(&wg)
-	wg.Wait()
+	tcpAddr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		fmt.Print(err)
+	}
+	tcpServer, err := net.ListenTCP("tcp", tcpAddr)
+	if err != nil {
+		fmt.Print(err)
+	}
+	defer tcpServer.Close()
 
 	s := &Snmp{
-		Agents: []string{"tcp://127.0.0.1:56789"},
+		Agents: []string{fmt.Sprintf("tcp://%s", tcpServer.Addr())},
 		ClientConfig: snmp.ClientConfig{
 			Translator: "netsnmp",
 		},
 	}
-	err := s.Init()
+	err = s.Init()
 	require.NoError(t, err)
```

Comment:
You can avoid that situation by directly do
```suggestion
	require.NoError(t, s.Init())
```